### PR TITLE
feat: add secret controller

### PR DESCRIPTION
**Pull request by randalljohnson**
Add a secret controller to manage secrets in AWS Secrets Manager on behalf of Threeport users.

### CODE REVIEW

File: internal/secret/secret_definition_config.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	// Create a Secrets Manager awssmClient
+	awssmClient := secretsmanager.NewFromConfig(*awsConfig)
+
+	// Define the secret name and value
+	var secretData map[string]string
+	if err = json.Unmarshal([]byte(*c.secretDefinition.Data), &secretData); err != nil {
+		return fmt.Errorf("failed to unmarshal secret data")
+	}
+
+	// encrypt sensitive values
```
Comment (randalljohnson) [inline internal/secret/secret_definition_config.go:73]: update comment

File: internal/secret/secret_instance.go [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+	}
+
+	// configure secret json manifests
+	jsonManifests, err := c.configureSecretControllerJsonManifests()
+	if err != nil {
+		return 0, fmt.Errorf("failed to configure secret controller json manifests: %w", err)
+	}
+
+	// update workload with secret manifests
+	switch c.workloadInstanceType {
```
Comment (randalljohnson) [inline internal/secret/secret_instance.go:74]: why not just check which instance ID is nil?

File: internal/secret/secret_instance.go [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+// secretInstanceCreated reconciles state for a secret instance
+// instance when it is removed.
+func secretInstanceDeleted(
+	r *controller.Reconciler,
+	secretInstance *v0.SecretInstance,
+	log *logr.Logger,
+) (int64, error) {
+	return 0, nil
+}
```
Comment (randalljohnson) [inline internal/secret/secret_instance.go:153]: implement

File: internal/workload/util/util.go [hunk 4]
```diff
@@ -115,7 +115,8 @@ func GetUniqueWorkloadResourceInstance(workloadResourceInstances *[]v0.WorkloadR
 			return nil, fmt.Errorf("failed to unmarshal json: %w", err)
 		}
 
-		if mapDef["kind"] == kind {
+		if mapDef["kind"] == kind &&
+			wri.ScheduledForDeletion == nil {
```
Comment (randalljohnson) [inline internal/workload/util/util.go:119]: Should we need this?

File: pkg/api-server/v1/tagged_fields_gen.go [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// generated by 'threeport-sdk codegen api-version' - do not edit
+
+package v1
+
+import v0 "github.com/threeport/threeport/pkg/api-server/v0"
+
+var AttachedObjectReferenceTaggedFields = make(map[string]*v0.FieldsByTag)
+var WorkloadInstanceTaggedFields = make(map[string]*v0.FieldsByTag)
```
Comment (randalljohnson) [inline pkg/api-server/v1/tagged_fields_gen.go:8]: This isn't referenced by anything, but it's still being generated.

File: pkg/client/v0/attached_object.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+	}
+
+	decoder := json.NewDecoder(bytes.NewReader(jsonData))
+	decoder.UseNumber()
+	if err := decoder.Decode(&attachedObjectReferences); err != nil {
+		return nil, fmt.Errorf("failed to decode object in response data from threeport API: %w", err)
+	}
+
+	return &attachedObjectReferences, nil
+}
```
Comment (randalljohnson) [inline pkg/client/v0/attached_object.go:87]: These are not currently referenced by anything.

File: cmd/database-migrator/migrations/00006_update_attached_object_references.go [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+func Up00006(ctx context.Context, db *sql.DB) error {
+	// create new columns
+	if _, err := db.Exec(`
+		ALTER TABLE attached_object_references
+		ADD COLUMN attached_object_type varchar(255),
+		ADD COLUMN attached_object_id bigint,
+		ADD COLUMN object_type varchar(255);
+	`); err != nil {
+		return err
+	}
```
Comment (randalljohnson) [inline cmd/database-migrator/migrations/00006_update_attached_object_references.go:26]: Use a transaction context here.

Comment (randalljohnson) [reply cmd/database-migrator/migrations/00006_update_attached_object_references.go:26]: Test transactions with intentional error mid-way through this function.

File: pkg/kube/v1/namespace.go [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+)
+
+// SetNamespaces adds the namespace resource and namespace assignment as needed
+// to an array of workload resource instances.
+func SetNamespaces(
+	workloadResourceInstances *[]v0.WorkloadResourceInstance,
+	workloadInstanceName *string,
+	workloadInstanceID *uint,
+	discoveryClient *discovery.DiscoveryClient,
+) (*[]v0.WorkloadResourceInstance, error) {
```
Comment (randalljohnson) [inline pkg/kube/v1/namespace.go:22]: remove and update v0/namespace.go

**Review by HarisChoudhary — APPROVED** [review]
🚢

**Review by lander2k2 — APPROVED** [review]

**Pull request merged**